shell/common/shell.cc

```diff
@@ -859,9 +872,10 @@ void Shell::HandleEngineSkiaMessage(fml::RefPtr<PlatformMessage> message) {
   if (args == root.MemberEnd() || !args->value.IsInt())
     return;
 
+  raster_cache_max_bytes_user_value_ = args->value.GetInt();
```

Comment:
This is running on the UI thread, but `Shell::OnPlatformViewSetViewportMetrics` runs on the platform thread.  Use a `std::atomic` type for `raster_cache_max_bytes_user_value_`.

Or perhaps a better way to go would be holding `raster_cache_max_bytes_user_value_` in a `std::optional` member within the rasterizer, which will only be accessed on the GPU thread.  `SetResourceCacheMaxBytes` can take a flag indicating whether it should set the `raster_cache_max_bytes_user_value_`.

Reply:
Done